streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -40,12 +50,87 @@
 
     private static final Logger log = LoggerFactory.getLogger(Stores.class);
 
+    public static KeyValueBytesStoreSupplier persistentKeyValueStore(final String name) {
+        return new RocksDbKeyValueBytesStoreSupplier(name);
+    }
+
+    public static KeyValueBytesStoreSupplier inMemoryKeyValueStore(final String name) {
+        return new KeyValueBytesStoreSupplier() {
+            @Override
+            public String name() {
+                return name;
+            }
+
+            @Override
+            public KeyValueStore<Bytes, byte[]> get() {
+                return new InMemoryKeyValueStore<>(name, Serdes.Bytes(), Serdes.ByteArray());
+            }
+
+            @Override
+            public String metricsScope() {
+                return "in-memory-state";
+            }
+        };
+    }
+
+    public static KeyValueBytesStoreSupplier lruMap(final String name, final int maxCacheSize) {
+        return new KeyValueBytesStoreSupplier() {
+            @Override
+            public String name() {
+                return name;
+            }
+
+            @Override
+            public KeyValueStore<Bytes, byte[]> get() {
+                return new MemoryNavigableLRUCache<>(name, maxCacheSize, Serdes.Bytes(), Serdes.ByteArray());
+            }
+
+            @Override
+            public String metricsScope() {
+                return "in-memory-lru-state";
+            }
+        };
+    }
+
+    public static WindowBytesStoreSupplier persistentWindowStore(final String name,
+                                                                 final long retentionPeriod,
+                                                                 final int numSegments,
+                                                                 final long windowSize,
+                                                                 final boolean retainDuplicates) {
+        return new RocksDbWindowBytesStoreSupplier(name, retentionPeriod, numSegments, windowSize, retainDuplicates);
```

Comment:
nit single parameter per line

Reply:
i think these ones are ok - the are all within a reasonable line length